Remove turn_on and turn_off feature for clients

ehendrix23 opened the pull request:
## Description:

Remove listing turn_on and turn_off as a supported feature for DVR clients as power_on and power_off do not work and are not supported in the DirecTV API for clients. It is only supported for the main DVR.

## Checklist:
  - [X] The code change is tested and works locally.
  - [X] Local tests pass with `tox`. **Your PR cannot be merged unless tests pass**

ehendrix23 committed:
Enhancements for DirecTV media player

Following enhancements have been made:

1. Added debug logging
2. Added ability to change channel using select_source service of the remote platform.
3. State will now show paused if a recorded program is paused, for live TV playing will always be returned.
4. Added the following attributes:
    a. media_position: current position of the media (in seconds)
    b. media_position_updated_at: timestamp when media_position was updated.
   c. source: current source (channel).
   d. media_isbeingrecorded: if current media is being recorded or not.
   e. media_rating: TV/Movie rating of the media
   f. media_recorded: if current media is recorded or live TV
   g. media_starttime: Timestamp media was aired

Reordered properties to follow same order as how they are in __init__.py of remote platform.

ehendrix23 committed:
Fixed error and cleaned up few items

Fixed an issue when determining if a program is recorded or not.
Cleaned up some coding.

ehendrix23 committed:
Added available property

Added available property

ehendrix23 committed:
Disable feature TURN_ON and TURN_OFF for DVR clients

Disable the feature turn_on and turn_off for DVR clients.

balloob commented on homeassistant/components/media_player/directv.py:
```diff
@@ -126,10 +130,15 @@ def __init__(self, name, host, port, device):
         self._paused = None
         self._last_position = None
         self._is_recorded = None
+        self._is_client = False if device == '0' else True
```
`device != '0'`

balloob commented on homeassistant/components/media_player/directv.py:
```diff
@@ -327,13 +336,15 @@ def media_start_time(self):
         """Turn on the receiver."""
-        _LOGGER.debug("Turn on %s", self._name)
-        self.dtv.key_press('poweron')
+        if not self._is_client:
```
This check should raise an error as it should never be called.

ehendrix23 replied:
@balloob, what kind of error should it raise? Is there something in HASS then to call for the error?

balloob replied:
`raise NotImplementedError()`

ehendrix23 committed:
self._is_client and raise NotImplementedError

Updated setting self._is_client
Raise NotImplementedError if turn_on or turn_off is called for clients.

ehendrix23 commented:
@balloob, FYI not sure what to do here with this check. It failed on:

`homeassistant/helpers/device_registry.py:45:4: R1705: Unnecessary "elif" after "return" (no-else-return)`

But that is not something I changed. Travis also ran successful on my branch.
